src/views/product-tutorials-view/components/product-view-content/types.ts

```diff
@@ -19,47 +15,17 @@ interface FeaturedStackBlock extends Omit<FeaturedStackProps, 'children'> {
 	blocks: CardListBlock[]
 }
 
-interface BrandedCalloutBlock extends BrandedCalloutProps {
-	type: 'BrandedCallout'
-}
-
-interface LogoCardBlock extends Pick<LogoCardListItem, 'logo'> {
-	type: 'LogoCard'
-	/** A single collection identifier string, which will be filled in
-	 * using fetched inlineCollections data */
-	collectionSlug: string
-}
-
-type CardListBlock = {
+interface CardListBlock extends LogoCardListProps {
 	type: 'CardList'
-	items: LogoCardBlock[]
-}
-
-interface TutorialsStackBlock
-	extends Omit<TutorialsStackProps, 'featuredTutorials'> {
-	type: 'TutorialsStack'
-	/** Tutorial identifier strings, which will be filled in
-	 * using fetched inlineCollections data */
-	tutorialSlugs: string[]
-}
-
-interface CollectionsStackBlock
-	extends Omit<CollectionsStackProps, 'featuredCollections'> {
-	type: 'CollectionsStack'
-	/** Collection identifier strings, which will be filled in
-	 * using fetched inlineCollections data */
-	collectionSlugs: string[]
 }
 
 export type ProductViewBlock =
 	| FeaturedStackBlock
-	| BrandedCalloutBlock
 	| CardListBlock
-	| TutorialsStackBlock
-	| CollectionsStackBlock
+	| ({ type: 'BrandedCallout' } & BrandedCalloutProps)
+	| ({ type: 'TutorialsStack' } & TutorialsStackProps)
+	| ({ type: 'CollectionsStack' } & CollectionsStackProps)
```

Comment:
Props here are simplified a bit, since we format props server-side so can pass them more directly to components.